components/TreeHeatmap/index.js

```diff
@@ -119,16 +124,20 @@ export default class TreeHeatmap extends VisComponent {
 
     let keys = Object.keys(this.data[0]);
     let idColumn = this.idColumn;
-    if (!idColumn) {
+    if (idColumn === undefined) {
       idColumn = keys.includes('_id') ? '_id' : idColumn;
       idColumn = keys.includes('_') ? '_' : idColumn;
       idColumn = keys.includes('') ? '' : idColumn;
     }
+    if (idColumn === undefined) {
+      console.log('TreeHeatmap: No suitable idColumn found.');
+      return;
+    }
 
     let rows = [];
     let reachedMetadata = false;
     this.data.forEach(row => {
-      let id = row[idColumn];
+      let id = '' + row[idColumn];
```

Comment:
I'm going to support the `'' + (value)` to cast to string technique.  `String()` may be clearer to some, but it is 15x slower (https://jsperf.com/cast-to-string/2), so it is a bad practice to get into.  Empty string concatenation has been the de-facto string cast for long enough to be clear.